src/components/QueryForm.tsx

```diff
@@ -0,0 +1,248 @@
+      const queryParams = new URLSearchParams();
+  
+      setQueryParam('node_url', node, queryParams);
+      queryParams.set('min_age', minAge ?? '');
+      queryParams.set('max_age', maxAge ?? '');
+      setQueryParam('sex', sex, queryParams);
+      setQueryParam('diagnosis', isControl ? null : diagnosis, queryParams);
+      queryParams.set('is_control', isControl ? 'true' : '');
+      queryParams.set('min_num_sessions', minNumSessions ?? '');
+      setQueryParam('assessment', assessmentTool, queryParams);
+      setQueryParam('image_modal', imagingModality, queryParams);
+  
+      // Notes:
+      // 1. Deleting elements in an array as we loop over it is not good, either make a new object or filter (same thing)
+      // 2. using forEach on the QueryParams object, 
+      // 3. Do the filtering first / switch before adding 
+      // Solution:
+      // Push the keys to be deleted inside keysToDelete and loop over them and delete them from queryParams afterwards
+      const keysToDelete: string[] = [];
+  
+      queryParams.forEach((value, key) => {
+        // if All option is selected for nodes field, delete all node_urls
+        if (value === '' || value === 'allNodes') {
+          keysToDelete.push(key);
+        }
+      });
+  
+      keysToDelete.forEach((key) => {
+        queryParams.delete(key);
+      });
+  
+      return `${apiQueryURL}${queryParams.toString()}`;
+    }
+  
+    return (
+      <div className={isFederationAPI ? 'grid grid-cols-2 grid-rows-8 gap-2' : 'grid grid-cols-2 grid-rows-7 gap-2'}>
+        {
+        isFederationAPI && (
+        <div className='col-span-2'>
+          <CategoricalField 
+          label='Neurobagel graph'
+          options={nodeOptions.map((n) => ({
+            label: n.NodeName,
+            id: n.ApiURL
+          }))}
+          onFieldChange={(label, value) => updateCategoricalQueryParams(label, value)} 
+          multiple
+          inputValue={node}
+          />
+        </div>
+          )
+        }
+        <div className={isFederationAPI && 'row-start-2'}>
+          <ContinuousField
+            label="Min age"
+            onFieldChange={(label, value) => updateContinuousQueryParams(label, value)}
+          />
+        </div>
+        <div className={isFederationAPI && 'row-start-2'}>
+          <ContinuousField
+            label="Max age"
+            onFieldChange={(label, value) => updateContinuousQueryParams(label, value)}
+          />
+        </div>
+        <div className='col-span-2'>
+          <CategoricalField
+            label="Sex"
+            options={Object.entries(sexes).map(([key, value]) => ({
+              label: key,
+              id: value,
+            }))}
+            onFieldChange={(label, value) => updateCategoricalQueryParams(label, value)}
+            inputValue={sex}
+          />
+        </div>
+        <div className={isFederationAPI ? 'col-span-2 row-start-4' : 'col-span-2 row-start-3'}>
```

Comment:
Yea since we don't want to render the `node field` when the query tool is talking to a non-federation API, the grid needs to be adjusted and therefore the classes are set conditionally.
There may be a better way of doing this but since we decided not to fixate on the UI layout for now, I decided to go with the simplest solution that preserves functionality for the time being.